components/BarcodeScanner/index.js

```diff
@@ -0,0 +1,81 @@
+import { useRef, useState } from "react";
+import useSWR from "swr";
+import styled from "styled-components";
+import Image from "next/image";
+import dynamic from "next/dynamic";
+const Scanner = dynamic(() => import("@/components/Scanner"), {
+  nossr: true,
+});
+
+const StyledSection = styled.section`
+  position: relative;
+  overflow: hidden;
+`;
+
+const StyledCanvas = styled.canvas`
+  position: absolute;
+  top: 0px;
+  height: 100%;
+`;
+
+export default function BarcodeScanner() {
+  const [scanning, setScanning] = useState(true);
+  const [result, setResult] = useState(null);
+
+  const { data } = useSWR(
+    result ? `https://de.openfoodfacts.org/api/v0/product/${result}.json` : null
+  );
+
+  const scannerRef = useRef(null);
+
+  function handleDetected(_result) {
+    setResult(_result.codeResult.code);
+    setScanning(false);
+  }
+
+  return (
+    <>
+      {result && <p>{result}</p>}
+      {/*Scanner Logic optimizable? */}
+      {scanning ? (
+        <>
+          <StyledSection ref={scannerRef}>
+            <StyledCanvas className="drawingBuffer" width="640" height="480" />
```

Comment:
dominik and me  trieded to figure it out but we did not get it it is part of the scanner and when you remove this the scanner is broken

Reply:
but even the drawing buffer is doing something when I delete it it seems like some non visible frame of the cam screen or something is getting bigger so that the navigation is getting more space to the camscreen